### PR TITLE
feat: add support for Across2.5 SpokePool contracts

### PR DESCRIPTION
**Pull request by amateima**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,4 +86,29 @@ export class ExampleModule {
     return module;
   }
 }
-```
\ No newline at end of file
+```
+
+## Queuing mechanism
+
+```mermaid
```
Comment (amateima) [inline README.md:93]: Add a chart describing how the queues flow looks like
[See here](https://github.com/across-protocol/scraper-api/tree/amatei/acx-949-update-scraper-api-to-handle-new-spoke#queuing-mechanism)

File: migrations/1682328242098-RefundRequestedEvent.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "transactionIndex" integer NOT NULL, 
+        "address" character varying NOT NULL, 
+        "chainId" integer NOT NULL, 
+        "transactionHash" character varying NOT NULL, 
+        "logIndex" integer NOT NULL, 
+        "args" jsonb NOT NULL, 
+        "createdAt" TIMESTAMP NOT NULL DEFAULT now(), 
+        "updatedAt" TIMESTAMP NOT NULL DEFAULT now(), 
+        CONSTRAINT "UK_refund_requested_ev_transactionHash_logIndex" UNIQUE ("transactionHash", "logIndex"), CONSTRAINT "PK_fe4e8e0dc6a7229e92f317f646b" PRIMARY KEY ("id"))
+    `);
```
Comment (amateima) [inline migrations/1682328242098-RefundRequestedEvent.ts:21]: Add the table definition for storing RefundRequested events

File: src/modules/configuration/index.ts [hunk 3]
```diff
@@ -44,34 +49,49 @@ export const configValues = () => ({
...
-        startBlockNumber: 28604263,
-      },
+      [ChainIds.mainnet]: [
+        {
+          address: "0x4D9079Bb4165aeb4084c526a32695dCfd2F77381",
+          startBlockNumber: 14819486,
+          abi: JSON.stringify(EthereumSpokePool2Abi),
+          acrossVersion: "2",
+        },
+      ],
```
Comment (amateima) [inline src/modules/configuration/index.ts:62]: Modify the config so as multiple spoke pool contracts can be supported for each chain

**Review by dohaki — COMMENTED** [review]
Looks good 👍 Left some questions

File: src/modules/web3/model/filled-relay-ev.entity.ts [hunk 4]
```diff
@@ -48,7 +72,7 @@ export class FilledRelayEv {
   logIndex: number;
 
   @Column({ type: "jsonb" })
-  args: FilledRelayEventArgs;
+  args: FilledRelay2EvArgs | FilledRelay2_5EvArgs;
```
[the remark this answers is not in the document]
Comment (amateima) [reply src/modules/web3/model/filled-relay-ev.entity.ts:75]: The version can be determined based on the contract address of the event

File: src/modules/configuration/index.ts [hunk 5]
```diff
@@ -44,34 +49,49 @@ export const configValues = () => ({
...
-      },
-      [ChainIds.polygon]: {
-        address: "0x69B5c72837769eF1e7C164Abc6515DcFf217F920",
-        startBlockNumber: 28604263,
-      },
+      [ChainIds.mainnet]: [
+        {
+          address: "0x4D9079Bb4165aeb4084c526a32695dCfd2F77381",
+          startBlockNumber: 14819486,
+          abi: JSON.stringify(EthereumSpokePool2Abi),
```
[the remark this answers is not in the document]
Comment (amateima) [reply src/modules/configuration/index.ts:59]: I'm not using the factory provided by the contracts-v2 anymore, but instantiating the contracts directly, so the abi is needed
[Change here](https://github.com/across-protocol/scraper-api/pull/225/files#diff-ee213e0d6094502a8c6df45e92aa288c7ed2516ed253c459ac7f9879215f57c5R125)

[the remark this answers is not in the document]
Comment (dohaki) [reply src/modules/configuration/index.ts:59]: I was just wondering if we need to `JSON.stringify` the ABI instead of just passing it directly

**Review by dohaki — APPROVED** [review]

**Pull request merged**